app/livechat/client/views/app/tabbar/customerChatHistory.js

```diff
@@ -0,0 +1,230 @@
+	this.modifiedHistory = new ReactiveVar([])
+	this.history = new ReactiveVar([]);
+	this.offset = new ReactiveVar(0);
+	this.total = new ReactiveVar(0);
+	this.isAllChat = new ReactiveVar(true);
+	this.isChatClicked = new ReactiveVar(true);
+	this.autorun(async () => {
+		const { room } = await APIClient.v1.get(`rooms.info?roomId=${ currentData.rid }`);
+		if (room && room.v) {
+			this.visitorId.set(room.v._id);
+		}
+	});
+
+	this.autorun(async () => {
+		
+		if (!this.visitorId.get() || !currentData || !currentData.rid) {
+			return;
+		}
+
+		const offset = this.offset.get();
+	
+		this.isLoading.set(true);
+		const { history, total } = await APIClient.v1.get(`livechat/visitors.chatHistory/room/${ currentData.rid }/visitor/${ this.visitorId.get() }?count=${ ITEMS_COUNT }&offset=${ offset }`);
+		
+		this.isLoading.set(false);
+		this.total.set(total);
+		this.history.set(this.history.get().concat(history));
+		// this will load all the chat history chat messages
+		allHistoryChat = [];
+		var limit = 100;
+		let count = this.history.get();
+		if(allHistoryChat.length>0){
+			return
+		}
+		for(i=0;i<count.length;i++){
+			let id = count[i]._id;
+			let token = count[i].v.token;
+			setTimeout(async function(){
+			const historyResult  =  await APIClient.v1.get(`livechat/messages.history/${ id }?token=${token}&limit=${limit}`);
+			let count2 = historyResult.messages;
+			
+			for(j=0; j<count2.length; j++){
+
+					allHistoryChat.push(count2[j]);
+				
+			}
+			
+		},1000)
+			
+		}
+		
+	
+	});	
+});
+
+Template.customerChatHistory.events({
+	'scroll .visitor-scroll': _.throttle(function(e, instance) {
+		if (e.target.scrollTop >= (e.target.scrollHeight - e.target.clientHeight)) {
+			const history = instance.history.get();
+			if (instance.total.get() <= history.length) {
+				return;
+			}
+			return instance.offset.set(instance.offset.get() + ITEMS_COUNT);
+		}
+	}, 200),
+	'keyup #searchInput': async function(event,template){ 
+		
+		searchResults = [];
+		let text = event.target.value;
+		template.isChatClicked.set(false);
+
+		Session.set('found',false);
+		if(event.target.value == ''){
+			template.isAllChat.set(true);
+		}else{
+			template.isAllChat.set(false);
+			Template.instance().searchResult = new ReactiveVar([]);
+			
+			for(var i=0; i<allHistoryChat.length;i++){
+				if(allHistoryChat[i].msg == text){ // check search input matches with any msg
+					Session.set('found',true);
+					var d = Date.parse(allHistoryChat[i].ts);
+					var D = new Date();
+					D.setTime(d);
+					var newTs = D.toTimeString();
+				
+					var time = getTime(newTs);
+					allHistoryChat[i].time = time;
+					searchResults.push(allHistoryChat[i]); 
+					Session.set('searchResult',searchResults);
+					template.searchResult.set(searchResults)
+				
+				}
+				
+			}
+			
+			if(searchResults.length > 0){
+				Session.set('found',true);
+			
+			}
+			
+	}},
+	'click .list-chat': async function(event,template){
```

Comment:
Why do you need this method?
I didn't get it.

Reply:
i am getting the Room Id and token and storing into the session and getting from the customer chat history messages